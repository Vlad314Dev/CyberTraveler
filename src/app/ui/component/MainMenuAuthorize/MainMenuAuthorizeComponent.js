import './MainMenuAuthorizeStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import {
    AUTHORIZE_LOGIN_OPTION,
    AUTHORIZE_SIGNUP_OPTION,
    BACK_OPTION,
    optionLabels
} from 'UIStore/MainMenu/MainMenuConfig';

class MainMenuAuthorizeComponent extends PureComponent
{
    static propTypes = {
        activeOption: PropTypes.string.isRequired,
        setActiveOption: PropTypes.func.isRequired
    };

    setActiveOption = (optionName) => () => {
        const { setActiveOption } = this.props;
        setActiveOption(optionName);
    }

    getActiveOptionLabel()
    {
        const { activeOption } = this.props;
        const option = optionLabels.find((data) => data.key === activeOption);

        return option.label;
    }

    render()
    {
        return (
            <div block="MainMenuOption" elem="List">
                <span block="MainMenuOption" elem="ActiveLabel">{ this.getActiveOptionLabel() }</span>
                {optionLabels.map((option, index) => {
                    if (option.key === AUTHORIZE_LOGIN_OPTION
                        || option.key === AUTHORIZE_SIGNUP_OPTION
                        || option.key === BACK_OPTION 
                    ) {
                        return <div 
                            block="MainMenuOption" 
                            elem="Wrapper" 
                            key={ index }
                        >
                            <span 
                                block="MainMenuOption" 
                                elem="Value"
                                key={ index }
                                onClick={ this.setActiveOption(option.key) }
                            >{ option.label }</span>
                        </div>
                    }
                    
                    return null;
                })}
            </div>
        );
    }
}

export default MainMenuAuthorizeComponent;
