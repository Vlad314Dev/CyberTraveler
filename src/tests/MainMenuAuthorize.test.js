/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import MainMenuAuthorizeComponent from '../app/ui/component/MainMenuAuthorize/MainMenuAuthorizeComponent';

describe('MainMenuAuthorize', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <MainMenuAuthorizeComponent 
                activeOption={ 'AUTHORIZE_OPTION' }
                setActiveOption={ jest.fn() }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <MainMenuAuthorizeComponent 
                activeOption={ 'AUTHORIZE_OPTION' }
                setActiveOption={ jest.fn() }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
