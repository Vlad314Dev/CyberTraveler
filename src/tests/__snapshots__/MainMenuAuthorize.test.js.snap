// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuAuthorize Should render layout 1`] = `
Array [
  <div
    className="MainMenuOption__List"
  >
    <span
      className="MainMenuOption__ActiveLabel"
    >
      Authorize
    </span>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Login
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Sign Up
      </span>
    </div>
    <div
      className="MainMenuOption__Wrapper"
    >
      <span
        className="MainMenuOption__Value"
        onClick={[Function]}
      >
        Back
      </span>
    </div>
  </div>,
]
`;
